lc-hax/Scripts/Components/CharacterMovement.cs

```diff
@@ -26,6 +26,11 @@ class CharacterMovement : MonoBehaviour {
     KeyboardMovement? NoClipKeyboard { get; set; } = null;
     CharacterController? CharacterController { get; set; }
 
+    IController? EnemyController => PossessionMod.Instance?.Controller;
+
+    EnemyAI? ControlledEnemy => PossessionMod.Instance?.PossessedEnemy;
```

Comment:
This is bad. `CharacterMovement` is meant to be generic but now you've coupled it with Possession. I don't like this idea.

Reply:
I don't think we should ever expose enemy controller.

Reply:
is still generic, just if these two aren't null, they will work together, else it works as the same as normal.

Reply:
Functionally perhaps. Semantically, it's not.

Reply:
> Functionally perhaps. Semantically, it's not.

what you would suggest? im still fixing the enemy interaction by clicking e, im not finished with this, adjusting every interact collision distance and making sure is always a hit when it touches a door.
/

Reply:
Everything that has to do with Possession should stay in `PossessionMod` as best as you can.

Reply:
would you prefer me using a bool like @D1GQ  did?

Reply:
I think the way we were doing it was fine.

Reply:
> I am not sure how D1GQ did it but I think the way we were doing it was fine. Not ideal, but at least it's SRP.

he put a accessible bool [`a4add73` (#419)](https://github.com/winstxnhdw/lc-hax/pull/419/commits/a4add73f44c29a6d941da647b6e947b84ea14e18)

Reply:
Oh yeah, that's fine as well.

Reply:
Ok, reverted